network/tls.go

```diff
@@ -0,0 +1,171 @@
+package network
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"errors"
+	"time"
+
+	"fmt"
+
+	"github.com/dedis/cothority/log"
+)
+
+/*
+Implementation of a TLS-host to allow for tls-connections. Only the server
+will authenticate to the client.
+*/
+
+// NewTLSRouter returns a new Router using TLSHost as the underlying Host.
+func NewTLSRouter(sid *ServerIdentity, key TLSKeyPEM) (*Router, error) {
+	h, err := NewTLSHost(sid, key)
+	if err != nil {
+		return nil, err
+	}
+	r := NewRouter(sid, h)
+	return r, nil
+}
+
+// TLSHost implements the Host interface using TLS connections.
+type TLSHost struct {
+	*TLSListener
+	addr Address
+	cert TLSCertPEM
+	key  TLSKeyPEM
+}
+
+// NewTLSHost returns a new Host using TLS connection based type.
+func NewTLSHost(si *ServerIdentity, key TLSKeyPEM) (*TLSHost, error) {
+	h := &TLSHost{
+		addr: si.Address,
+		cert: si.Cert,
+		key:  key,
+	}
+	var err error
+	h.TLSListener, err = NewTLSListener(si, key)
+	return h, err
+}
+
+// Connect establishes a connection to the remote host given in addr.
+func (h *TLSHost) Connect(si *ServerIdentity) (Conn, error) {
+	log.Lvl3(h.addr, "connecting to", si.Address)
+	addr := si.Address
+	switch addr.ConnType() {
+	case TLS:
```

Comment:
Do we really want to have a `TLSHost` accept `PlainTCP` connections? I know we said so, but it seems insecure and an easy attack, no?